crates/bevy_reflect/bevy_reflect_derive/src/from_reflect.rs

```diff
@@ -191,34 +221,76 @@ fn get_active_fields(
                     field.reflection_index.expect("field should be active"),
                     is_tuple,
                 );
-                let ty = field.data.ty.clone();
+                let ty = field.reflected_type().clone();
+                let real_ty = &field.data.ty;
 
                 let get_field = quote! {
                     #bevy_reflect_path::#struct_type::field(#dyn_struct_name, #accessor)
                 };
 
+                let into_remote = |value: proc_macro2::TokenStream| {
+                    if field.attrs.is_remote_generic().unwrap_or_default() {
+                        quote! {
+                            #FQOption::Some(
+                                // SAFETY: The remote type should always be a `#[repr(transparent)]` for the actual field type
+                                unsafe {
+                                    ::core::mem::transmute_copy::<#ty, #real_ty>(
+                                        &::core::mem::ManuallyDrop::new(#value?)
+                                    )
```

Comment:
It's been a while so I don't quite remember 😅 

I'll test it out again and leave a comment above this line based on my findings. I think it had to do with lifetimes or something, but yeah I'll check again.

Reply:
yeah that's what i was worried about. i'm not 100% sure but i'm always wary of `transmute_copy` especially because it can unsoundly extend/shorten lifetimes!

Reply:
Updated with a comment. Turns out it had to do with the compiler being unable to determine that generic types have the same size.